documentation/API-authentication.md

```diff
@@ -0,0 +1,50 @@
+# Authentication in the API's
+> Note: there is discussion how we can support NL Edukoppeling for SaaS Vendor https://github.com/stichtingsem/ecosystem/issues/9
+> So the rest of this document is prelimenary until that is settled
+
+## Authentication
+For authentication of API calls we will initially use auth2 tokens in the rest calls.
+
+So before calling the API, the API client needs to call the authentication server to get a authentication token to do the request. The token will contain the identity of the party calling the API, one or more scope's (see [scopes](##authorization-&-Scope)) 
+
+This is an industry standard (See https://oauth.net/2/) and is supported by many software languages and frameworks.
+
+
+![architecture](diagrams/API-Flows.drawio.svg)
+
+For more details: see [API-authentication-token](API-authentication-token.md)
+
+
+# School identity
+Many calls are made on behalf of the school (after consent). The school will be incuded as claim in the oauth token.
+
+For each call the api endpoint knows which party did the call and on behalf of which school
+
+
+> To be discussed: Until now the preference is the schoolidentity is part of the token, so the authentication server can do the check if the school has give consent to process the data n behalf of the school
+> 
+> Alternative is that the schoolidentity becomes part of the data or part of the url or that there is an endpoint per school
+
+For more details: see [API-authentication-token](API-authentication-token.md)
+
+
+
+# Authorization & Scopes
+Each API documentation describes what scope is needed in the authentication token. These are described in the [API reference](//reference/README.md)
```

Comment:
Should it not be the set of claim(s) that is in the token? As I understand it, the scopes are used during the initial flow to indicate which information (claims) is requested in the token.

So to me this would result in:
- intiating the authentication flow with a SemConsents Scope
- resulting in a JWT token with the given consents in (a) claim(s)  (so then the service can check the claims to see if the call is allowed).